zubhub_backend/zubhub/projects/views.py

```diff
@@ -586,3 +591,63 @@ def delete(self, request, *args, **kwargs):
                 profile.save()
 
             return result
+
+
+class ViolationReasonsListAPIView(ListAPIView):
+    """
+    List all violation reasons.
+
+    Requires authentication and staff or moderator.
+    Returns violations reasons.
+    """
+
+    queryset = ViolationReason.objects.all()
+    serializer_class = ViolationReasonSerializer
+    permission_classes = [IsAuthenticated, IsStaffOrModerator]
+    throttle_classes = [CustomUserRateThrottle, SustainedRateThrottle]
+
+
+class ProjectViolationsClearApiView(APIView):
+    """
+    Clear all violations for a project
+
+    Requires authentication and owner.
+    Returns project details.
+    """
+
+    permission_classes = [IsAuthenticated, IsOwner]
+    throttle_classes = [CustomUserRateThrottle, SustainedRateThrottle]
+
+    def post(self, request, *, pk):
+        try:
+            old = Project.objects.get(pk=pk)
+        except Project.DoesNotExist:
+            return Response('Not found', status=404)
+
+        if not request.user or old.creator.id != request.user.id:
+            return Response('Error', status=403)
+
+        old.violations.set([])
+        old.save()
+        return Response('Success')
+
+
+class ProjectViolationAddApiView(CreateAPIView):
+    """
+    Creates a new violation and adds it to a project.
+
+    Requires authentication and staff or moderator.
+    Returns the violation details.
+    """
+    queryset = Violation.objects.all()
+    serializer_class = ViolationSerializer
+    permission_classes = [IsAuthenticated, IsStaffOrModerator]
+    throttle_classes = [PostUserRateThrottle, SustainedRateThrottle]
+
+    def perform_create(self, serializer):
+        pk = self.kwargs.get("pk")
+        project = Project.objects.get(id=pk)
+        violation = serializer.save(creator=self.request.user, project=project)
+        project.violations.add(violation)
+        project.save()
+        return violation
```

Comment:
Send notification to user to let them know their project was unpublished.

Reply:
Does this mean we need a new notification type?

Reply:
Yeah